packages/flame/lib/src/experimental/layout_component.dart

```diff
@@ -0,0 +1,143 @@
+import 'package:flame/components.dart';
+import 'package:flame/experimental.dart';
+import 'package:flame/src/game/notifying_vector2.dart';
+import 'package:flutter/widgets.dart';
+
+enum Direction { horizontal, vertical }
+
+/// Super class for [RowComponent] and [ColumnComponent]
+/// A relayout is performed when
+///  - a change in this component's children takes place
+///  - the size of this component or a parent changes
+///  - the [gap] parameter is changed
+abstract class LayoutComponent extends PositionComponent with HasGameRef {
+  LayoutComponent(
+    this.direction,
+    this.mainAxisAlignment,
+    this._gap,
+    this.isManuallySized,
```

Comment:
I think this should take a `size` too and automatically set `isManuallySized` to true if it is provided and then omit the boolean for the constructor, but having it as mutable so that the user can change it afterwards.